client/asset/btc/txdb.go

```diff
@@ -105,29 +113,75 @@ func NewBadgerTxDB(filePath string, log dex.Logger) (*BadgerTxDB, error) {
 		return nil, err
 	}
 
-	return &BadgerTxDB{
+	seq, err := db.GetSequence([]byte("seq"), 10)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, die := context.WithCancel(context.Background())
+
+	badgerDB := &BadgerTxDB{
 		DB:  db,
-		log: log}, nil
+		log: log,
+		seq: seq,
+		wg:  new(sync.WaitGroup),
+		die: die,
+	}
+	badgerDB.running.Store(true)
+
+	badgerDB.wg.Add(1)
+	go func() {
+		defer badgerDB.wg.Done()
+		ticker := time.NewTicker(5 * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				err := db.RunValueLogGC(0.5)
+				if err != nil && !errors.Is(err, badger.ErrNoRewrite) {
+					log.Errorf("garbage collection error: %v", err)
+				}
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return badgerDB, nil
+}
+
+// badgerDB returns ErrConflict when a read happening in a update (read/write)
+// transaction is stale. This function retries updates multiple times in
+// case of conflicts.
+func handleConflictWithBackoff(update func() error) error {
```

Comment:
This is interesting. I thought the Badger was ACID and the precluded the need for this type of weirdness. But reading [this blog entry](https://dgraph.io/blog/post/badger-txn/), I'm not sure. Did you actually see `ErrConflict` in use?